### PR TITLE
NEW (DevOps) @W-15619715@ Added GHAs for more automated release process.

### PR DESCRIPTION
**Pull request by jfeingold35**

### CODE REVIEW

File: .github/workflows/create-github-release.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Checkout main
+        uses: actions/checkout@v4
+        with:
+          ref: main
+      - name: Get version property
+        id: get-version-property
+        run: |
+          PACKAGE_VERSION=$(jq -r ".version" package.json)
+          echo "package_version=$PACKAGE_VERSION" >> "$GITHUB_OUTPUT"
+      - name: Download VSIX artifact
```
Comment (jfeingold35) [inline .github/workflows/create-github-release.yml:40]: The VSIX gets attached to the release, and this is how we'll do that.

File: .github/workflows/create-release-branch.yml [hunk 2]
```diff
@@ -50,7 +50,7 @@ jobs:
     permissions:
       contents: write
     outputs:
-      branch-name: ${{ steps.create-branch.branch_name }}
+      branch-name: ${{ steps.create-branch.outputs.branch_name }}
```
Comment (jfeingold35) [inline .github/workflows/create-release-branch.yml:53]: Fixing a typo.

File: SHA256-template.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+   https://code.visualstudio.com/docs/editor/extension-gallery#_common-questions.
+   For example, download,
+   https://salesforce.gallery.vsassets.io/_apis/public/gallery/publisher/salesforce/extension/salesforcedx-vscode-core/57.15.0/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage.
+
+2. From a terminal, run:
+
+shasum -a 256 <location_of_the_downloaded_file>
+
+3. Confirm that the SHA in your output matches the value in this list of SHAs.
+   SHA_TEMPLATE
```
Comment (jfeingold35) [inline SHA256-template.md:18]: Identical to the actual `SHA256.md` except for this template string; lets us use `sed` to easily update the SHA programatically.

File: github-actions/verify-pr-title/package.json [hunk 4]
```diff
@@ -17,14 +17,13 @@
...
         "bundle": "npm run format:write && npm run package",
         "ci-test": "npx jest",
-        "coverage": "npx make-coverage-badge --output-path ./badges/coverage.svg",
         "format:write": "npx prettier --write .",
         "format:check": "npx prettier --check .",
         "lint": "npx eslint src/**/*.ts",
         "package": "npx ncc build src/index.ts -o dist --source-map --license licenses.txt",
         "package:watch": "npm run package -- --watch",
         "test": "npx jest",
-        "all": "npm run format:write && npm run lint && npm run test && npm run coverage && npm run package"
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/package.json:27]: The coverage command no longer makes any sense and was causing a compile failure, so I removed it.

File: github-actions/verify-pr-title/__tests__/verifyMain2DevPrTitle.test.ts
Comment (jfeingold35) [inline github-actions/verify-pr-title/__tests__/verifyMain2DevPrTitle.test.ts:1]: Updated tests.

File: SHA256-template.md (hunk 3 above)
Comment (stephen-carter-at-sf) [reply SHA256-template.md:18]: Can we use a variable name that says something like SHA_PLACEHOLDER. Or better yet, make it actually look like a placeholder or variable with something like: <<SHA_VALUE>>

**Review by stephen-carter-at-sf — APPROVED** [review]

File: SHA256-template.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+   following the instructions at
+   https://code.visualstudio.com/docs/editor/extension-gallery#_common-questions.
+   For example, download,
+   https://salesforce.gallery.vsassets.io/_apis/public/gallery/publisher/salesforce/extension/salesforcedx-vscode-core/57.15.0/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage.
+
+2. From a terminal, run:
+
+shasum -a 256 <location_of_the_downloaded_file>
+
+3. Confirm that the SHA in your output matches the value in this list of SHAs.
```
Comment (jag-j) [inline SHA256-template.md:17]: Did we not have have a template file until this point? Nit - can we move this to a different directory (may be under templates/) so users don't see this at the root?

**Review by jag-j — APPROVED** [review]

**Review by stephen-carter-at-sf — APPROVED** [review]

**Pull request merged**